tests/networkpolicy_test.go

```diff
@@ -223,17 +287,70 @@ func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
 	}, 5*time.Second, 1*time.Second).ShouldNot(Succeed())
 }
 
-func waitForNetworkPolicyDeletion(policy *v1network.NetworkPolicy) {
+func createNetworkPolicy(namespace, name string, labelSelector metav1.LabelSelector, ingress []networkv1.NetworkPolicyIngressRule) *networkv1.NetworkPolicy {
+	policy := &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: networkv1.NetworkPolicySpec{
+			PodSelector: labelSelector,
+			Ingress:     ingress,
+		},
+	}
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	By(fmt.Sprintf("Create networkpolicy %s/%s", policy.Namespace, policy.Name))
+	policy, err = virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Create(policy)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed creating network policy %s/%s", policy.Namespace, policy.Name))
+	return policy
+}
+
+func waitForNetworkPolicyDeletion(policy *networkv1.NetworkPolicy) {
```

Comment:
i think we should defer policy=nil in this function, this will make it trust-able and re-entrant

Reply:
Defer has nothing to do with re-entrant two threads can run it in parallel also policy pointer is passed by value you are not changing the pointer value from external valirable.

Reply:
re-entrant is not just for parallel, its also for using the same function again
if you dont nil it, the results wont be as expected

why passed by value, its a pointer isnt it ? cant we change the address the pointer points to ?

Reply:
At golang variables are passed by value so pointers are passed by value, and reentrant is all about multithreading, I am not going to add a defer here.

Reply:
https://stackoverflow.com/questions/2799023/what-exactly-is-a-reentrant-function

i saw in the past more uses of the word re-entrant, we can search them deeply but i dont see the need.
this link might show that its not the same thing / only use of the word.

and you can do like this 
https://stackoverflow.com/questions/42047889/change-structs-pointer-value-in-structs-method
in order to nil the ptr if we want to code to be right imo.

Reply:
I am **NOT** going to create a super structure with a pointer just for something that is clearly not needed, this is a helper function in a tests that work with the assumption that is called with a **proper** policy and is called **after all is done** and assumptions are not bad things for testing.

Reply:
you decide whatever you want, i just want to say that Edy thought as well its an interesting point, in other context.
its two different things to decide not to do this on this PR versus to accept its not needed to do it at all, at least from my side of view.

i do think its needed, else the function has limitations, so as long as we decide that we are fine with the limitations, i dont agree its not needed, again, if to do this in this PR is other story.

Reply:
Ok so I decide not to do it here, I am going to close this one.